hawk/src/main/java/com/orhanobut/hawk/Hawk.java

```diff
@@ -111,20 +102,90 @@ public static void init(Context context, String password, LogLevel logLevel) {
      * @return true if put is successful
      */
     public static <T> boolean put(String key, List<T> list) {
+        return put(key, null, list, true, true) != null;
+    }
+
+    /*
+     * A helper put method.
+     *
+     * @param key          the key
+     * @param value        the value if isList is false, otherwise null
+     * @param list         the list, if isList is true, otherwise null
+     * @param isList       determines if a list instance should be present
+     * @param addToStorage determines if the keyval should be immediately added to storage
+     */
+    private static <T> String put(String key, T value, List<T> list, boolean isList,
```

Comment:
I like to generalize the methods as well, but in this case we should keep the structure as it is. Because currently we only support objects, primitives and List. But for the future if we want to add other collection types such as Map and Set, we need to modify this method which I don't like modifying methods which has big impact to others. This is against the Open/Closed principle. Overloading has the minimum side effect to others.